### DISCUSSION TITLE
"Mark all messages as read" option in chat

### DISCUSSION THREAD

**Discussion by guillezcurra** [tomtau/superbasic-im#7] [Ideas]
Hello, sometimes we open a chat and only want to read the messages and not answer them, I think that where could be an option at the top of the chat (in chats.html) to "Mark as read" all unread messages at that specific chat, what do you think?

Thank you.

**Comment by tomtau** [comment]
Perhaps or should it automatically send that when a chat page is opened?

**Reply by guillezcurra** [reply]
I thought about that but maybe you want to read these messages in a computer later that day (because of attachments or being important info). This could ensure that you can take a look and forget about them until the evening, when you see those unread messages in the PC.
With the "Mark as read" option you can choose.

Of course this is your project! I am only sharing my ideas of how I would prefer it.

Thanks!

**Reply by tomtau** [reply]
That makes sense. Actually, I often use a similar workflow with leaving some e-mails unread to check them later, so it could work with instant messaging as well.

**Reply by guillezcurra** [reply]
Great! I think that this method is the way to do it:
https://docs.wwebjs.dev/Client.html#sendSeen

I can try to develop it!

**Reply by tomtau** [reply]
@guillezcurra yes, I think that's the method to use on the client. Alternatively, there's also a method on the Chat object: https://docs.wwebjs.dev/Chat.html#sendSeen but I assume it may be easier to use the one on the Client object.

(BTW There's also an inverse method `markChatUnread`, but I think it won't change the unread message count or send anything to the other chat parties.)

> I can try to develop it!

Sure, you're more than welcome to try! I opened this issue: https://github.com/tomtau/superbasic-im/issues/8
